API/Controllers/BooksController.cs

```diff
@@ -0,0 +1,101 @@
+{
+    [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme)]
+    [Route("api/vl")]
+    [ApiController]
+    public class BooksController : ControllerBase
+    {
+        private readonly IBookService _bookService;
+
+        public BooksController(IBookService bookService)
+        {
+            _bookService = bookService;
+        }
+
+        /// <summary>
+        /// Get all books
+        /// </summary>
+        /// <returns></returns>
+        [HttpGet, Authorize("IsAllUsers")]
+        [ProducesResponseType(typeof(bool), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status400BadRequest)]
+        [ProducesDefaultResponseType]
+        public async Task<IActionResult> GetBooks()
+        {
+            var books = await _bookService.GetBooksAsync();
+            if (books == null) return BadRequest();
+            return Ok(books);
+        }
+
+        /// <summary>
+        /// Get a specific book by id
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        [HttpGet("{id}"), Authorize("IsAllUsers")]
+        public async Task<ActionResult<Book>> GetBook(int id)
+        {
+            var book = await _bookService.GetBookAsync(id);
+            if (book == null) return BadRequest();
+            return Ok(book);
+        }
+
+        /// <summary>
+        /// Edit a book
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="book"></param>
+        /// <returns></returns>
+        [HttpPut("{id}"), Authorize("IsLibrarian")]
+        [ProducesResponseType(typeof(bool), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status400BadRequest)]
+        [ProducesDefaultResponseType]
+        public async Task<IActionResult> UpdateBook(Book book)
+        {
+            var bookUpdate = await _bookService.UpdateBookAsync(book);
+            return Ok(bookUpdate);
+        }
+
+        /// <summary>
+        /// Create a book
+        /// </summary>
+        /// <param name="book"></param>
+        /// <returns></returns>
+        [HttpPost, Authorize("IsLibrarian")]
+        public async Task<ActionResult<Book>> CreateBook(Book book)
+        {
+            var newBook = await _bookService.CreateBookAsync(book);
+            if (newBook == null) return BadRequest();
+            return Ok(newBook);
+        }
+
+        /// <summary>
+        /// Delete a book by id
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        [HttpDelete("{id}"), Authorize("IsLibrarian")]
+        [ProducesResponseType(typeof(bool), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status400BadRequest)]
+        [ProducesDefaultResponseType]
+        public async Task<IActionResult> DeleteBook(int id)
+        {
+            var result = await _bookService.DeleteBookAsync(id);
+
+            if (result)
```

Comment:
Resharper should be indicating that this can be flattened

if(!result) {
return BadRequest();
}

return Ok();

Reply:
![image](https://user-images.githubusercontent.com/99216796/195893957-a52af4e0-e0fb-4b4a-90c5-516c1463e96d.png)

Reply:
It does not

Reply:
Look at the `else` see how its lightly colored, there is a suggestion to refactor when you click / hover over that